equinox/nn/_attention.py

```diff
@@ -144,12 +148,14 @@ def __init__(
         output_size: Optional[int] = None,
         qk_size: Optional[int] = None,
         vo_size: Optional[int] = None,
+        state_length: Optional[int] = None,
         use_query_bias: bool = False,
         use_key_bias: bool = False,
         use_value_bias: bool = False,
         use_output_bias: bool = False,
         dropout_p: float = 0.0,
         inference: bool = False,
+        kv_cache: Optional[AbstractKVCache] = None,
```

Comment:
This argument doesn't seem to be documented in the docstring.

Reply:
Also, I think it would be really useful to have a short example somewhere -- maybe at the bottom of the `__init__` docstring? Not a strong feeling on location -- on how to use `MultiheadAttention` with state and kv caching.

Reply:
Also I think `state_length` is unused here now? It appears solely in the `kv_cache`. We should remove it as an argument.